controllers/apps/transformer_cluster_normalization.go

```diff
@@ -409,3 +410,51 @@ func (t *clusterNormalizationTransformer) writeBackCompNShardingSpecs(transCtx *
 		transCtx.Cluster.Spec.Shardings = shardings
 	}
 }
+
+func (t *clusterNormalizationTransformer) patchCRDAPIVersionKey(transCtx *clusterTransformContext) error {
+	apiVersions := map[string][]string{}
+
+	add := func(k, v string) {
+		if _, ok := apiVersions[k]; !ok {
+			apiVersions[k] = []string{}
+		}
+		apiVersions[k] = append(apiVersions[k], v)
+	}
+
+	from := func(name string, annotations map[string]string) {
+		if annotations == nil {
+			add("", name)
+		} else {
+			add(annotations[constant.CRDAPIVersionAnnotationKey], name)
+		}
```

Comment:
For a map type, if you only get an object, GoLang will return a zero value of the corresponding value type even if the map is nil.

Reply:
Done